crates/swc_ecma_minifier/src/compress/optimize/iife.rs

```diff
@@ -322,6 +296,96 @@ where
         clean_params(callee);
     }
 
+    /// If a parameter is not used, we can ignore return value of the
+    /// corresponding argument.
+    pub(super) fn ignore_unused_args_of_iife(&mut self, e: &mut CallExpr) {
+        let has_spread_arg = e.args.iter().any(|v| v.spread.is_some());
+        if has_spread_arg {
+            return;
+        }
+
+        if !self.options.unused {
+            return;
+        }
+
+        let callee = match &mut e.callee {
+            Callee::Super(_) | Callee::Import(_) => return,
+            Callee::Expr(e) => &mut **e,
+        };
+
+        match find_body(callee) {
+            Some(body) => match body {
+                Either::Left(body) => {
+                    if contains_arguments(body) {
+                        return;
+                    }
+                }
+                Either::Right(body) => {
+                    if contains_arguments(body) {
+                        return;
+                    }
+                }
+            },
+            None => return,
+        }
+
+        if let Expr::Fn(FnExpr {
+            ident: Some(ident), ..
+        }) = callee
+        {
+            if self
+                .data
+                .vars
+                .get(&ident.to_id())
+                .filter(|usage| usage.used_recursively)
+                .is_some()
+            {
+                return;
+            }
+        }
+
+        let mut removed = vec![];
+        let params = find_params(callee);
+        if let Some(mut params) = params {
+            // We check for parameter and argument
+            for (idx, param) in params.iter_mut().enumerate() {
+                if let Pat::Ident(param) = &mut **param {
+                    if let Some(usage) = self.data.vars.get(&param.to_id()) {
```

Comment:
Does `data.vars` track all scopes? I'm assuming `to_id` generates a globally unique value for every binding?

Reply:
Yes, `data.vars` is whole data of the Program and all `Id` is unique in swc.

You can read more about it at https://rustdoc.swc.rs/swc_ecma_transforms_base/fn.resolver.html and https://rustdoc.swc.rs/swc_ecma_ast/struct.Ident.html